src/components/tree-item/tree-item.tsx

```diff
@@ -0,0 +1,291 @@
+  //PROPS
+  /**
+   * Set this attribute if you want the gxg-treeitem to display a checkbox
+   */
+  @Prop() checkbox = false;
+
+  /**
+   * Set this attribute if you want the gxg-treeitem checkbox to be checked by default
+   */
+  @Prop() checked = false;
+
+  /**
+   * Set this attribute when you are downloading a resource
+   */
+  @Prop() downloading = false;
+
+  /**
+   * Set thhe left side icon from the available Gemini icon set : https://gx-gemini.netlify.app/?path=/story/icons-icons--controls
+   */
+  @Prop() leftIcon: string;
+  /**
+   
+
+  /**
+   * Set thhe right side icon from the available Gemini icon set : https://gx-gemini.netlify.app/?path=/story/icons-icons--controls
+   */
+  @Prop() rightIcon: string;
+  /**
+
+  /**
+   * If this tree-item has a nested tree, set this attribute to make the tree open by default
+   */
+  @Prop() open = false;
+
+  //STATE
+  @Prop() isLeaf: boolean;
+  @State() paddingLeft = "4px";
+  @State() numberOfParentTrees = 1;
+  @State() height = 0;
+  @State() parentHasCheckbox = false;
+  @State() hasParentTree = false;
+  @State() itemPaddingLeft;
+  @State() verticalLineHeight: string;
+  @State() horizontalLinePaddingLeft = 0;
+  @State() hidePlusMinusIcon: boolean;
+  @State() showDownloadingIcon = false;
+
+  //EVENTS
+  @Event() itemToggled: EventEmitter;
+
+  //LISTEN
+  @Listen("itemToggled")
+  itemToggledHandler() {
+    this.calculateItemHeight();
+  }
+
+  @Element() el: HTMLElement;
+
+  componentWillLoad() {
+    //If tree item has not a tree-item inside, is leaf
+    const treeItemHasTree = this.el.querySelector('[slot="tree"]');
+    if (this.isLeaf === undefined) {
+      if (treeItemHasTree === null) {
+        this.isLeaf = true;
+      }
+    }
+
+    //If has parent tree
+    const parentElementNodeName = this.el.parentElement.nodeName;
+    if (parentElementNodeName === "GXG-TREE") {
+      this.hasParentTree = true;
+    }
+  }
+
+  componentDidLoad() {
+    //Count number of parent trees in order to set the apporpiate padding-left
+    this.numberOfParentTrees = this.getParents(this.el);
+    //Get the item height
+    this.calculateItemHeight();
+    //If parent has checkbox...
+    const parentHasCheckbox = this.el.parentElement.getAttribute("checkbox");
+    if (parentHasCheckbox !== null) {
+      this.parentHasCheckbox = true;
+    }
+
+    //Defines the height of the vertical line that associates the chid items with the parent item
+    setTimeout(
+      function() {
+        this.calculateVerticalLineHeight();
+      }.bind(this),
+      500
+    );
+  }
+
+  @Watch("downloading")
+  downloadingHandler() {
+    if (this.downloading) {
+      this.hidePlusMinusIcon = true;
+      setTimeout(
+        function() {
+          this.showDownloadingIcon = true;
+        }.bind(this),
+        500
+      );
+    } else {
+      this.showDownloadingIcon = false;
+      setTimeout(
+        function() {
+          this.hidePlusMinusIcon = false;
+        }.bind(this),
```

Comment:
Consider using an arrow function here:

```suggestion
        ()  => this.hidePlusMinusIcon = false,
```

Reply:
Great suggestion, thanks.